### PR TITLE
Xuejun/remove api in ov any

### PR DESCRIPTION
**Pull request by zhaixuejun1993**
### Details:
 - remove interface: operator T&()
 - remove interface: operator const T&() const&
 - remove interface: operator T&() const& 
 - remove interface: Base* operator->()
 - remove interface: const Base* operator->() const
 - remove interface: Base* get()
 - remove interface: const Base* get() const
 - remove interface: static std::shared_ptr<T> call(const Any& any)
 - remove interface: std::shared_ptr<T> dynamic_pointer_cast(const ::ov::Any& any)
 - remove interface: std::shared_ptr<T> static_pointer_cast(const ::ov::Any& any)

### Tickets:
 - 100091

### CODE REVIEW

File: src/core/tests/any.cpp [hunk 1]
```diff
@@ -419,15 +419,15 @@ TEST_F(AnyTests, constructFromVariantImpl) {
 TEST_F(AnyTests, dynamicPointerCastToVariantWrapper) {
     OPENVINO_SUPPRESS_DEPRECATED_START
     Any p = std::make_shared<ngraph::VariantWrapper<std::string>>("42");
-    auto str_variant = std::dynamic_pointer_cast<ngraph::VariantWrapper<std::string>>(p);
+    auto str_variant = ov::Any(p).as<std::shared_ptr<ngraph::VariantWrapper<std::string>>>();
```
Comment (ilya-lavrenov) [inline src/core/tests/any.cpp:422]: are we going to remove VariantWrapper itself?
it's also deprecated. Maybe we need to start with it?

File: tools/legacy/benchmark_app/main.cpp [hunk 2]
```diff
@@ -292,7 +292,7 @@ int main(int argc, char* argv[]) {
                 if (isFlagSetInCommandLine("nthreads"))
                     device_config[GNA_CONFIG_KEY(LIB_N_THREADS)] = std::to_string(FLAGS_nthreads);
             } else {
-                std::vector<std::string> supported_config_keys = ie.GetMetric(device, METRIC_KEY(SUPPORTED_CONFIG_KEYS));
+                std::vector<std::string> supported_config_keys = (ie.GetMetric(device, METRIC_KEY(SUPPORTED_CONFIG_KEYS))).as<std::vector<std::string>>();
```
Comment (ilya-lavrenov) [inline tools/legacy/benchmark_app/main.cpp:295]: why do we need extra `()` ?

File: src/core/tests/any.cpp [hunk 3]
```diff
@@ -419,15 +419,15 @@ TEST_F(AnyTests, constructFromVariantImpl) {
...
     ASSERT_EQ("42", str_variant->get());
```
Comment (riverlijunjie) [inline src/core/tests/any.cpp:423]: These test items are flagged with "OPENVINO_SUPPRESS_DEPRECATED_START", do we need keep them?

File: src/plugins/intel_gna/src/ops/util/util.hpp [hunk 4]
```diff
@@ -166,7 +166,7 @@ inline bool is_power_activation(const ov::Node* node) noexcept {
             return true;
         }
         return (1.0f != value);
-    } else if (auto power_op = std::dynamic_pointer_cast<ngraph::op::PowerIE>(node)) {
+    } else if (auto power_op = ov::Any(node).as<std::shared_ptr<ngraph::op::PowerIE>>()) {
```
Comment (ilya-lavrenov) [inline src/plugins/intel_gna/src/ops/util/util.hpp:169]: looks strange that we need to wrap to ov::Any first.
Why do we need to change something here?

File: src/inference/src/dev/converter_utils.cpp [hunk 5]
```diff
@@ -273,7 +273,7 @@ class IInferencePluginWrapper : public InferenceEngine::IInferencePlugin {
     }
 
     void SetCore(std::weak_ptr<InferenceEngine::ICore> core) override {
-        return m_plugin->set_core(std::dynamic_pointer_cast<ov::ICore>(core));
+        return m_plugin->set_core(ov::Any(core).as<std::shared_ptr<ov::ICore>>());
```
Comment (ilya-lavrenov) [inline src/inference/src/dev/converter_utils.cpp:276]: why?

Comment (riverlijunjie) [inline src/inference/src/dev/converter_utils.cpp:276]: Why change it?

File: src/inference/src/dev/core_impl.cpp [hunk 6]
```diff
@@ -204,7 +204,7 @@ ov::Plugin ov::CoreImpl::get_plugin(const std::string& pluginName) const {
             allowNotImplemented([&]() {
                 // Add device specific value to support device_name.device_id cases
                 std::vector<std::string> supportedConfigKeys =
-                    plugin.get_property(METRIC_KEY(SUPPORTED_CONFIG_KEYS), {});
+                    plugin.get_property(METRIC_KEY(SUPPORTED_CONFIG_KEYS), {}).as<std::vector<std::string>>();
```
Comment (riverlijunjie) [inline src/inference/src/dev/core_impl.cpp:207]: Confused, removing deprecated api will cause code become redundant.
Is it expected?

File: src/core/tests/any.cpp (hunk 3 above)
Comment (ilyachur) [reply src/core/tests/any.cpp:423]: I think we need to remove `VariantWrapper` which is deprecated and in this case we will be able to remove `OPENVINO_SUPPRESS_DEPRECATED_START`

File: src/inference/src/dev/core_impl.cpp (hunk 6 above)
Comment (ilyachur) [reply src/inference/src/dev/core_impl.cpp:207]: We can use `auto supportedConfigKeys = ...` :)

Comment (zhaixuejun1993) [reply src/inference/src/dev/core_impl.cpp:207]: > We can use `auto supportedConfigKeys = ...` :)

If we change to `auto supportedConfigKeys = ...`, the type of 'supportedConfigKeys' is 'ov::Any'. The using of 'supportedConfigKeys' is inconvenient, such as the next line' supportedConfigKeys.begin()' no member 'begin()'

File: src/plugins/intel_gna/src/ops/util/util.hpp (hunk 4 above)
Comment (zhaixuejun1993) [reply src/plugins/intel_gna/src/ops/util/util.hpp:169]: > looks strange that we need to wrap to ov::Any first. Why do we need to change something here?
**std::dynamic_pointer_cast**<ngraph::op::PowerIE>(node)) used here is not the standards func from std, which is is the api (deprecated) defined in Any. Maybe we can convert it by dynamic_case like this 
`auto power_op = dynamic_cast<const ngraph::op::PowerIE*>(node)`
How do u think?

File: src/inference/src/dev/converter_utils.cpp (hunk 5 above)
Comment (zhaixuejun1993) [reply src/inference/src/dev/converter_utils.cpp:276]: > why?
How about this?
```suggestion
        return m_plugin->set_core(std::dynamic_pointer_cast<ov::ICore>(core.lock()));
```

File: src/core/tests/any.cpp (hunk 1 above)
Comment (zhaixuejun1993) [reply src/core/tests/any.cpp:422]: can ignore this, which will be removed in PR: 15580

File: src/inference/src/dev/converter_utils.cpp (hunk 5 above)
Comment (zhaixuejun1993) [reply src/inference/src/dev/converter_utils.cpp:276]: > Why change it?

@riverlijunjie  std::dynamic_pointer_cast used here is not the standards func from std, which is is the api (deprecated) defined in Any.

File: src/inference/src/dev/core_impl.cpp (hunk 6 above)
Comment (zhaixuejun1993) [reply src/inference/src/dev/core_impl.cpp:207]: > We can use `auto supportedConfigKeys = ...` :)

Got it! Will change

File: src/plugins/intel_gna/src/ops/util/util.hpp [hunk 7]
```diff
@@ -166,7 +166,7 @@ inline bool is_power_activation(const ov::Node* node) noexcept {
             return true;
         }
         return (1.0f != value);
-    } else if (auto power_op = std::dynamic_pointer_cast<ngraph::op::PowerIE>(node)) {
+    } else if (auto power_op = dynamic_cast<const ngraph::op::PowerIE*>(node)) {
```
Comment (ilya-lavrenov) [inline src/plugins/intel_gna/src/ops/util/util.hpp:169]: can we keep unchanged?

Comment (zhaixuejun1993) [reply src/plugins/intel_gna/src/ops/util/util.hpp:172]: > can we keep unchanged?

If keep unchanged, this api `std::shared_ptr<T> dynamic_pointer_cast(const ::ov::Any& any)` can not be deprecated.

**Review by ilya-lavrenov — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Comment by riverlijunjie** [comment]
Looks like we need always explicitly specify data type for ov::Any?

**Comment by zhaixuejun1993** [comment]
Need this PR for fixing CUDA plugin error
https://github.com/openvinotoolkit/openvino_contrib/pull/613

**Pull request merged**